bokehjs/src/coffee/util/util.coffee

```diff
@@ -51,7 +51,35 @@ get_indices = (data_source) ->
   else
     []
 
+map_vector_that_may_contain_patches_with_holes = (vector, map) ->
+  # Possible inputs are:
+  # [1, 2, 3]
+  # [1, 2, NaN, 3, 4]
+  # [[[1, 2], [3, 4]]]
+  # [1, 2, NaN, [[1, 2], [3, 4]]]
+  # Note: At the top level items are either a single value or an array of arrays
+  result = []
```

Comment:
@bryevdv - regarding your comment elsewhere about Float64Arrays. The place to add them back is here. This function will get messy, but at least it's not duplicated anymore so hopefully contains the mess.